frontend/src/utils/retryWrapper.js

```diff
@@ -6,7 +6,8 @@ export const retryWrapper = (p, timeout, retryN) =>
     p()
       .then(resolve)
       .catch(e => {
-        if (retryN === MAX_RETRIES) return reject(e);
+        const retryable = e?.response?.status !== 400;
+        if (retryN === MAX_RETRIES || !retryable) return reject(e);
```

Comment:
Small improvement on the overall requests handler. No point on retrying 400 requests.